### PR TITLE
boards: mimxrt101x_evk: add boot and slot partitions

### PR DESCRIPTION
**Pull request by butok**
Add boot and slot partitions for mimxrt1010_evk and mimxrt1015_evk.
Enable MCUBoot & Flash support for them.

### CODE REVIEW

**Review by decsny — APPROVED** [review]

**Review by danieldegrasse — CHANGES_REQUESTED** [review]
This PR won't work correctly until we override the `BOOT_MAX_IMG_SECTORS` value for this board, since the flash partitions here are so large. For 7MiB partitions, we will want a max sector count of (0x700000 / 0x1000) = 1792 or better.

Here is an example of a PR where I did this for the RT595: https://github.com/mcu-tools/mcuboot/pull/1583.

Once the PR is merged upstream, you should either ping `nordicjm` or `nvlsianpu`, who can synchronize Zephyr's copy of MCUBoot with the upstream copy to pull in the relevant changes.

File: boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts [hunk 1]
```diff
@@ -20,11 +20,13 @@
 	};
 
 	chosen {
-		zephyr,sram = &dtcm;
+		zephyr,sram = &ocram;
```
[the remark this answers is not in the document]
Comment (butok) [reply boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts:23]: 32KB of RAM is not enough for MCUBoot, have to use OCRAM.

**Comment by butok** [comment]
> This PR won't work correctly until we override the `BOOT_MAX_IMG_SECTORS` value for this board, since the flash partitions here are so large. For 7MiB partitions, we will want a max sector count of (0x700000 / 0x1000) = 1792 or better.
> 
> Here is an example of a PR where I did this for the RT595: [mcu-tools/mcuboot#1583](https://github.com/mcu-tools/mcuboot/pull/1583).
> 
> Once the PR is merged upstream, you should either ping `nordicjm` or `nvlsianpu`, who can synchronize Zephyr's copy of MCUBoot with the upstream copy to pull in the relevant changes.

Sure. The separate MCUBoot commit is targeting this issue: https://github.com/mcu-tools/mcuboot/pull/1868

File: boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts (hunk 1 above)
[the remark this answers is not in the document]
Comment (butok) [reply boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts:23]: BTW: OCRAM is 64KB, DTCM is 32KB ;)

**Comment by danieldegrasse** [comment]
> Sure. The separate MCUBoot commit is targeting this issue: https://github.com/mcu-tools/mcuboot/pull/1868

Thanks, @nordicjm would you be willing to update the MCUBoot revision used with Zephyr to pull in the changes in the upstream MCUBoot PR?

**Comment by butok** [comment]
> > Sure. The separate MCUBoot commit is targeting this issue: [mcu-tools/mcuboot#1868](https://github.com/mcu-tools/mcuboot/pull/1868)
> 
> Thanks, @nordicjm would you be willing to update the MCUBoot revision used with Zephyr to pull in the changes in the upstream MCUBoot PR?

Here: https://github.com/zephyrproject-rtos/zephyr/pull/65855#pullrequestreview-1753225390

**Review by danieldegrasse — APPROVED** [review]

**Comment by danieldegrasse** [comment]
@butok I'm going to add the [DNM] tag to this PR for now, you should be able to remove it once the MCUBoot synchronization PR merges. If not, please ping me.

**Comment by butok** [comment]
> @butok I'm going to add the [DNM] tag to this PR for now, you should be able to remove it once the MCUBoot synchronization PR merges. If not, please ping me.

OK. I will wait also for "rt1015: correct FlexRAM bank allocation" https://github.com/zephyrproject-rtos/zephyr/pull/65890 .

**Pull request merged**